From e3cb6d96aaacccd88404f1e01422d09a5ecdcb32 Mon Sep 17 00:00:00 2001
From: Matt Porter <mporter@ti.com>
Date: Mon, 17 Sep 2012 12:33:08 -0400
Subject: [PATCH 8/8] video: st7735fb: add pinctrl support

Add explicit pinctrl support since gpio-omap doesn't have support
for pinctrl in it.

Signed-off-by: Matt Porter <mporter@ti.com>
---
 drivers/video/st7735fb.c |    8 ++++++++
 1 files changed, 8 insertions(+), 0 deletions(-)

diff --git a/drivers/video/st7735fb.c b/drivers/video/st7735fb.c
index db9e83b..fb62bd8d3 100644
--- a/drivers/video/st7735fb.c
+++ b/drivers/video/st7735fb.c
@@ -21,10 +21,12 @@
 #include <linux/gpio.h>
 #include <linux/spi/spi.h>
 #include <linux/delay.h>
+#include <linux/err.h>
 #include <linux/uaccess.h>
 #include <linux/of.h>
 #include <linux/of_device.h>
 #include <linux/of_gpio.h>
+#include <linux/pinctrl/consumer.h>
 
 #include "st7735fb.h"
 
@@ -434,6 +436,7 @@ static int __devinit st7735fb_probe (struct spi_device *spi)
 	struct device_node *np = spi->dev.of_node, *child;
 	enum of_gpio_flags flags;
 	const struct spi_device_id *spi_id = spi_get_device_id(spi);
+	struct pinctrl *pinctrl;
 
 	if (!spi_id) {
 		dev_err(&spi->dev,
@@ -441,6 +444,11 @@ static int __devinit st7735fb_probe (struct spi_device *spi)
 		return -EINVAL;
 	}
 
+	pinctrl = devm_pinctrl_get_select_default(&spi->dev);
+	if (IS_ERR(pinctrl))
+		dev_warn(&spi->dev,
+			"pins are not configured from the driver\n");
+
 #ifdef __LITTLE_ENDIAN
 	vmem = (u8 *)vmalloc(vmem_size);
 #else
-- 
1.7.7.6

